File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string common(string s1, string s2){
        int n = min(s1.size(), s2.size());
        string s = "";
        for(int i=0; i<n; i++){
            if(s1[i] == s2[i]){
                s.push_back(s1[i]);
            }
            else{
                break;
            }
        }
        return s;
    }
    string longestCommonPrefix(vector<string>& strs) {
        string s = strs[0];
        for(int i=1; i<strs.size(); i++){
            s = common(s,strs[i]);
        }
        return s;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def common(self, s1: str, s2: str) -> str:
        s = ""
        n = min(len(s1), len(s2))
        for i in range(n):
            if s1[i] == s2[i]:
                s += s1[i]
            else:
                break
        return s

    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        s = strs[0]
        for i in range(1, len(strs)):
            s = self.common(s, strs[i])
            if not s:
                break
        return s